src/o365/graph/commands/groupsetting/groupsetting-set.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../GraphCommand';
+import { GroupSetting } from './GroupSetting';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+}
+
+class GraphGroupSettingSetCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.GROUPSETTING_SET}`;
+  }
+
+  public get description(): string {
+    return 'Updates the particular group setting';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = args.options.id;
+    return telemetryProps;
+  }
+
+  public allowUnknownOptions(): boolean | undefined {
+    return true;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        if (this.verbose) {
+          cmd.log(`Retrieving group setting with id '${args.options.id}'...`);
+        }
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/groupSettings/${args.options.id}`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          }),
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((groupSetting: GroupSetting): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(groupSetting);
+          cmd.log('');
+        }
+
+        const requestOptions: any = {
+          url: `${auth.service.resource}/v1.0/groupSettings/${args.options.id}`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none',
+            'content-type': 'application/json'
+          }),
+          body: {
+            displayName: groupSetting.displayName,
+            templateId: groupSetting.templateId,
+            values: this.getGroupSettingValues(args.options, groupSetting)
+          },
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.patch(requestOptions);
+      })
+      .then((res: any): void => {
+        if (this.debug) {
+          cmd.log('Response:')
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        cmd.log(res);
```

Comment:
The patch request to MSGraph doesn't return any response, so this line outputs `undefined` which can be removed.